### ISSUE TITLE
Theme name generation runs into error for some hostnames

### ISSUE THREAD

**Issue by we5** [Shopify/shopify-cli#1316]
# Issue summary

Write a short description of the issue here ↓

running command `shopify theme serve` results in an error

## Expected behavior

What do you think should happen?

Successfully serving the development theme

## Actual behavior

What actually happens?

Command results in the following error:

```
/usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/api.rb:78:in `block in request': 422 (ShopifyCli::API::APIRequestClientError)
{"errors":{"name":["ist zu lang (mehr als 50 Zeichen)"]}}
```

## Steps to reproduce the problem

1. run `shopify theme serve`
1. error comes up

## Specifications

- App type:
- Operating System: macOS Big Sur 11.4
- Shell: zsh
- Ruby version (ruby -v): 2.7.0

**Comment by we5** [comment]
I debugged this locally and was able to fix this by limiting the characters of the returned hostname through this line:
https://github.com/Shopify/shopify-cli/blob/0d5eb6141c2d21eaed3f53f5a09d5dbe27dfa557/lib/shopify-cli/theme/development_theme.rb#L58

**Comment by we5** [comment]
I added a pull request here:
#1317

**Comment by macournoyer** [comment]
Nice find! Thanks a lot for the PR Björn 🙌

**Comment by juergengenser** [comment]
any news on when this fix gets merged?

**Comment by we5** [comment]
Sorry for the delay. Unfortunately due to a surgery and high work load i was not pushing this on my side. Will re-create the pull-request this week

**Comment by unreact-tamenishi** [comment]
@we5 

I also get the same error when I try` shopify theme` serve. Did the error improve?


```
/usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/api.rb:78:in `block in request': 422 (ShopifyCli::API::APIRequestClientError)
{"errors":{"name":["ist zu lang (mehr als 50 Zeichen)"]}}

```


### Specifications
App type:
Operating System: macOS Big Sur 11.2.3
Shell: zsh
Ruby version (ruby -v): 3.02

**Comment by we5** [comment]
It depends on when the pull request gets merged and released.

**Comment by unreact-tamenishi** [comment]
@we5 

Thank you for your reply!

Is there any way to fix the error myself at this point?

**Comment by macournoyer** [comment]
Fixed in #1429
